kernel/src/scan/data_skipping.rs

```diff
@@ -242,7 +243,8 @@ impl DataSkippingFilter {
         // 3. The selection evaluator does DISTINCT(col(predicate), 'false') to produce true (= keep) when
         //    the predicate is true/null and false (= skip) when the predicate is false.
         let select_stats_evaluator = engine.get_expression_handler().get_evaluator(
-            stats_schema.clone(),
+            // safety: kernel is very broken if we don't have the schema for Add actions
+            get_log_schema().project(&[ADD_NAME]).unwrap(),
```

Comment:
This was incorrectly passing `stats_schema` as the input schema before. the input to this evaluator is the raw log data, so we want to add schema. it only worked before because we were ignoring the input schema

Reply:
This is the _input_ schema. This evaluator is called on the raw data read from the log, so we're asking for it to get `add.stats` out of the `Add`. `STATS_EXPR` is just `Expr::column("add.stats")`, and note the output is just a `String`.